test/Uri.js

```diff
@@ -48,8 +48,19 @@ describe('Uri', function() {
 		assert.strictEqual('', uri.getOrigin());
 	});
 
+	it('should set procotol http: on localhost uri', function() {
+		var uri = new Uri('localhost:8080');
+		assert.strictEqual('http:', uri.getProtocol());
+		assert.strictEqual('8080', uri.getPort());
+	});
+
+	it('should support all types of uri schemes', function() {
+		var uri = new Uri('test:hostname');
+		assert.strictEqual('test:', uri.getProtocol());
+	});
+
 	it('should support set port on uri', function() {
-		var uri = new Uri('hostname:8080');
```

Comment:
Yes, that is exactly why I removed the URL pattern above.  

Based on the new logic, the value before the `:`,  `hostname`, will be interpreted as the protocol and it will then be passed into the `URL()` constructor which will also interpret it in the same way.  

In the case where the `hostname`  is something other than `localhost` and it doesn't contain a `.`, the new logic will consider the value as a valid protocol and will not prepend `http://` to the uri.

The implementation in this PR will behave more similar to the standard that the `URL` constructor uses but it can still have unexpected behavior as in the case above.